modules/new_profile/jsx/NewProfileIndex.js

```diff
@@ -0,0 +1,257 @@
+      credentials: 'same-origin',
+      body: formObject,
+    })
+    .then((resp) => resp.json())
+    .then((data) => {
+     this.setState({newData: data});
+     this.setState({isCreated: true});
+    });
+                }
+    }
+/* eslint-enable no-console */
+
+    /**
+   * Set the form data based on state values of child elements/componenets
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding form element
+   */
+    setFormData(formElement, value) {
+        let formData = JSON.parse(JSON.stringify(this.state.formData));
+        formData[formElement] = value;
+
+        this.setState(
+            {
+                formData: formData,
+            }
+        );
+    }
+
+     render() {
+            let profile = null;
+            let edc = null;
+            let project = null;
+            let pscid = null;
+            if (this.state.configData['useProject'] === 'true') {
+                project =
+                <div>
+                 <SelectElement
+                   name ="project"
+                   label ="Project"
+                   options ={this.state.configData.project}
+                   onUserInput ={this.setFormData}
+                   ref ="project"
+                   value ={this.state.formData.project}
+                   required ={true}
+                 />
+                </div>;
+            }
+            if (this.state.configData['edc'] === 'true') {
+                edc =
+                 <div>
+                 <DateElement
+                   name ="edcDateTaken"
+                   label ="Expected Date of Confinement"
+                   minYear ="2000"
+                   maxYear ="2017"
+                   onUserInput ={this.setFormData}
+                   ref ="edcDateTaken1"
+                   value ={this.state.formData.edcDateTaken}
+                   required ={true}
+                 />
+                 <DateElement
+                   name ="edcDateTakenConfirm"
+                   label ="Confirm EDC"
+                   minYear ="2000"
+                   maxYear ="2017"
+                   onUserInput ={this.setFormData}
+                   ref ="edcDateTaken2"
+                   value ={this.state.formData.edcDateTakenConfirm}
+                   required ={true}
+                 />
+                 </div>;
+            }
+            if (this.state.configData['pscidSet'] === 'true') {
+                pscid =
+                <div>
+                  <TextboxElement
+                   name ="pscid"
+                   label ="PSCID"
+                   onUserInput ={this.setFormData}
+                   ref ="pscid"
+                   value ={this.state.formData.pscid}
+                   required ={true}
+                  />
+                </div>;
+            }
+            if (!this.state.isCreated) {
+                profile =
+                <FormElement
```

Comment:
might be nice to wrap the form in a panel element

Reply:
done